src/app.js

```diff
@@ -41,6 +42,35 @@ app.use(rewriteUnsupportedBrowserMethods);
 app.engine('handlebars', handlebarsInstance.engine);
 app.set('view engine', 'handlebars');
 
+app.use(
+  session({
+    name: 'AuthCookie',
+    secret: "Michael Karsen Loves Hoola Hooping",
+    saveUninitialized: true,
+    resave: false
+  })
+);
+
+app.use('*', (req, res, next) => {
+  //This will run every single time any route is called
+  let date = new Date().toUTCString();
+  let reqmethod = req.method;
+  let reqroute = req.originalUrl;
+  let loggedin = false;
+  if(req.session.user) loggedin = true;
+  //Logs some useful stuff to the console, can be commented out
+  console.log(`[${date}]: ${reqmethod} ${reqroute} | Authorized: ${loggedin}`);
+  //if you're logged in, proceed to whatever page you were trying to access
+  if(loggedIn) {
+    next();
+  }
+  //if you aren't logged in, you're redirected to the home page, never even hitting the page you were trying to hit
+  else {
+    res.redirect('/');
```

Comment:
I say we do 403 forbidden page instead

Reply:
Sounds good, wasn't sure how we wanted to do that